### DISCUSSION TITLE
Dynamic Packet Rate

### DISCUSSION THREAD

**Discussion by ss8688** [ExpressLRS/ExpressLRS#2360] [General]
Could we have have dynamic packet rate works like Dynamic Power? Decrease packet rate could increase range.I think it can reduce the probability of losing control

**Comment by FalkF** [comment]
I like the idea, this would let users reap the benefits how high rate connections when possible while at the same time getting the reliability of low rate modes.
Obviously there needs to be a clever algo making sure there is no bad interactions between dynamic power & packet rate (e.g. power scales down because connection is good and at same time rate scales up because, well, connection is good).
So we need to set a priority (low power, or high bandwith), or something like "increase power up to Xmw and try to achieve best rate, but if needed increase power further and drop rate".

Or is there maybe something fundamental preventing this (packet rate change could take a, possibliy noticeable, moment)?

Now that dynamic power works even better with 4.0 it might be a good feature.

**Reply by FilipeJrd** [reply]
My concern regarding this kind of feature is the latency variation that this kind of behavior would introduce. But since this could be a opt-in feature, I see no evil.
 
Maybe a decent and simple strategy is to start the packet rate down scaling once the max Tx power is reached. This would reduce the chances of weird feedback loops.

**Comment by pkendall64** [comment]
This is unlikely to happen as reconfiguring the transceiver takes some time and causes the link to drop.
It's even worse if the receiver is not capable of switching to the packet rate then the TX says to.
Especially now that we have Dual Band transmitters and they are compatible with single band receivers.
